compiler/rustc_hir_typeck/src/op.rs

```diff
@@ -521,6 +521,63 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
                         }
                     }
                 }
+
+                // Suggest using `add`, `offset` or `offset_from` for pointer - {integer},
+                // pointer + {integer} or pointer - pointer.
+                if op.span.can_be_used_for_suggestions() {
+                    if op.node == hir::BinOpKind::Add {
+                        match (lhs_ty.kind(), rhs_ty) {
+                            (RawPtr(..), rhs_ty) if rhs_ty.is_integral() => {
+                                let lhs_snip =
+                                    self.tcx.sess.source_map().span_to_snippet(lhs_expr.span);
+                                let rhs_snip =
+                                    self.tcx.sess.source_map().span_to_snippet(rhs_expr.span);
+                                if let Ok(lhs_snip) = lhs_snip && let Ok(rhs_snip) = rhs_snip {
+                                    err.span_suggestion(
+                                        expr.span,
+                                        "consider using `wrapping_add` or `add` for pointer + {integer}",
+                                        format!("{lhs_snip}.wrapping_add({rhs_snip})"),
+                                        Applicability::MaybeIncorrect
+                                    );
+                                }
+                            }
+                            _ => {}
+                        }
+                    } else if op.node == hir::BinOpKind::Sub {
+                        match (lhs_ty.kind(), rhs_ty) {
+                            (RawPtr(..), rhs_ty) if rhs_ty.is_integral() => {
+                                let lhs_snip =
+                                    self.tcx.sess.source_map().span_to_snippet(lhs_expr.span);
+                                let rhs_snip =
+                                    self.tcx.sess.source_map().span_to_snippet(rhs_expr.span);
+                                if let Ok(lhs_snip) = lhs_snip && let Ok(rhs_snip) = rhs_snip {
+                                    err.span_suggestion(
+                                        expr.span,
+                                        "consider using `offset` for pointer - {integer}",
```

Comment:
Since `ptr + {integer}` suggests `add` and `wrapping_add` instead of `offset`, should this similarly suggest `sub` and `wrapping_sub`?

Reply:
Yeah that makes sense, I'll change it